contribs/emissions/src/test/java/org/matsim/contrib/emissions/events/VehicleLeavesTrafficEventsTest.java

```diff
@@ -0,0 +1,95 @@
+package org.matsim.contrib.emissions.events;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.matsim.api.core.v01.Scenario;
+import org.matsim.contrib.emissions.EmissionModule;
+import org.matsim.contrib.emissions.VspHbefaRoadTypeMapping;
+import org.matsim.contrib.emissions.utils.EmissionsConfigGroup;
+import org.matsim.core.api.experimental.events.EventsManager;
+import org.matsim.core.config.Config;
+import org.matsim.core.config.ConfigUtils;
+import org.matsim.core.controler.AbstractModule;
+import org.matsim.core.controler.Injector;
+import org.matsim.core.events.EventsUtils;
+import org.matsim.core.events.MatsimEventsReader;
+import org.matsim.core.events.algorithms.EventWriterXML;
+import org.matsim.core.scenario.ScenarioUtils;
+import org.matsim.core.utils.io.IOUtils;
+import org.matsim.examples.ExamplesUtils;
+import org.matsim.utils.eventsfilecomparison.EventsFileComparator;
+
+/**
+ *
+ * Calculate offline emissions to test rare cases of vehicle
+ * enter and leave traffic events occurring on the same links.
+ *
+ * This class tests the (proper) handling of these cases as a result of
+ * the VehicleLeavesTrafficEvent method in the WarmEmissionHandler.
+ *
+ * @author Ruan J. Gräbe
+ */
+public class VehicleLeavesTrafficEventsTest {
+    private static final String hbefaColdFile = IOUtils.extendUrl( ExamplesUtils.getTestScenarioURL( "emissions-sampleScenario" ), "sample_41_EFA_ColdStart_vehcat_2020average.csv" ).toString();
+    private static final String hbefaWarmFile = IOUtils.extendUrl( ExamplesUtils.getTestScenarioURL( "emissions-sampleScenario" ), "sample_41_EFA_HOT_vehcat_2020average.csv" ).toString();
+    private static final String eventsFile =  "test/input/org/matsim/contrib/emissions/events/VehicleLeavesTrafficEventTest/smallBerlinSample.output_events.xml.gz";
+    private static final String vehiclesFile = "test/input/org/matsim/contrib/emissions/events/VehicleLeavesTrafficEventTest/smallBerlinSample_emissionVehicles.xml";
```

Comment:
Please use MatsimTestUtils.getInputDirectory() OR MatsimTestUtils.getClassInputDirectory() + < Filename > instead of defining the path manually "test/input.../"